### PR TITLE
Removes shortwave energy conservation fix

### PR DESCRIPTION
**Pull request by vanroekel**
Coupled testing showed unexpected extremely strong sensitivity to the
previous shortwave radiation energy fixer.  In that fix the excess
energy was used to heat the bottom layer.  This yielded a strong
response in tropical precipitation, dramatically increasing the double
ITCZ bias.  This removes that energy fixer and only tracks the solar
radiation that is discarded for extremely shallow cells

### CODE REVIEW

**Comment by vanroekel** [comment]
@mark-petersen and @jonbob this is the PR we need to go in for WC as soon as we possibly can.  It will be nonBFB with master.  It effectively removes the shortwave energy fixer as there were strong sensitivities to that fixer see here

https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.vanroekel/20210216.v2_newICs.lowerVisc.piControl.ne30pg2_EC30to60E2r2.chrysalis/e3sm_diags/180x360_aave/model_vs_obs_0021-0040/viewer/lat_lon/gpcp_v23/prect-global-gpcp_v23/ann.html

without the shortwave fixer

https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.golaz/E3SM/v2/beta/20210212.v2_newICs.piControl.ne30pg2_EC30to60E2r2.chrysalis/e3sm_diags/180x360_aave/model_vs_obs_0021-0040/viewer/lat_lon/gpcp_v23/prect-global-gpcp_v23/ann.html

@jonbob for E3SM, I would like to make sure `bottomLayerShortwaveTemperatureFlux` is added to the default `timeMonthly` stream for all resolutions

**Comment by vanroekel** [comment]
@maltrud if you have time to look at this PR, I'd appreciate it.

**Comment by vanroekel** [comment]
@mark-petersen is there time to add a quick change to this PR?  Chris Golaz requested adding a flag to enable adding the shortwave or not.  I can do that quickly if you are okay with that.

File: src/core_ocean/shared/mpas_ocn_tracer_short_wave_absorption_jerlov.F [hunk 1]
```diff
@@ -186,6 +190,23 @@ subroutine ocn_tracer_short_wave_absorption_jerlov_tend(meshPool, forcingPool, i
...
+            do k=1,maxLevelCell(iCell)
+               tend(index_temperature, k, iCell) = tend(index_temperature, k, iCell) + layerFraction
+            end do
+         end do
+#ifndef CPRPGI
+         !$omp end do
+         !$omp end parallel
+#endif
+      end if
+
```
Comment (vanroekel) [inline src/core_ocean/shared/mpas_ocn_tracer_short_wave_absorption_jerlov.F:209]: @mattdturner could you look at my openmp pragmas here?  I'm not sure of them

File: src/core_ocean/shared/mpas_ocn_tracer_short_wave_absorption_jerlov.F [hunk 2]
```diff
@@ -186,6 +190,23 @@ subroutine ocn_tracer_short_wave_absorption_jerlov_tend(meshPool, forcingPool, i
...
 #endif
 
+      if(config_enable_shortwave_energy_fixer) then
+#ifndef CPRPGI
+         !$omp parallel
+         !$omp do schedule(runtime) private(k, layerFraction)
+#endif
+         do iCell = 1, nCells
+            layerFraction = bottomlayerShortwaveTemperatureFlux(iCell) / (maxLevelCell(iCell) + 1.0E-15_RKIND)
+            do k=1,maxLevelCell(iCell)
```
Comment (vanroekel) [inline src/core_ocean/shared/mpas_ocn_tracer_short_wave_absorption_jerlov.F:200]: @mark-petersen does this seem safe against the future case of `maxLevelcell=0`?

File: src/core_ocean/shared/mpas_ocn_tracer_short_wave_absorption_variable.F [hunk 3]
```diff
@@ -196,6 +200,23 @@ subroutine ocn_tracer_short_wave_absorption_variable_tend(meshPool, swForcingPoo
...
+         do iCell = 1, nCells!
+            layerFraction = bottomlayerShortwaveTemperatureFlux(iCell) / (maxLevelCell(iCell) + 1.0E-15_RKIND)!
+            do k=1,maxLevelCell(iCell)!
+               tend(index_temperature, k, iCell) = tend(index_temperature, k, iCell) + layerFraction!
+            end do!
+         end do!
+#ifndef CPRPGI!
+         !$omp end do!
+         !$omp end parallel!
+#endif
```
Comment (vanroekel) [inline src/core_ocean/shared/mpas_ocn_tracer_short_wave_absorption_variable.F:217]: @mattdturner could you check this loop as well?

File: src/core_ocean/shared/mpas_ocn_tracer_short_wave_absorption_jerlov.F (hunk 1 above)
Comment (mattdturner) [reply src/core_ocean/shared/mpas_ocn_tracer_short_wave_absorption_jerlov.F:209]: This looks fine.

File: src/core_ocean/shared/mpas_ocn_tracer_short_wave_absorption_variable.F (hunk 3 above)
Comment (mattdturner) [reply src/core_ocean/shared/mpas_ocn_tracer_short_wave_absorption_variable.F:217]: This looks good as well

**Review by mattdturner — APPROVED** [review]
Approved via visual inspection.  With it being nonBFB, somebody more familiar with the expected results should test and approve as well.

**Comment by mark-petersen** [comment]
passes nighty regression with gnu, gnu debug, intel debug.

File: src/core_ocean/shared/mpas_ocn_tracer_short_wave_absorption_jerlov.F (hunk 2 above)
Comment (mark-petersen) [reply src/core_ocean/shared/mpas_ocn_tracer_short_wave_absorption_jerlov.F:200]: Yes.

**Review by mark-petersen — APPROVED** [review]
Yes, this looks good to me.

**Pull request merged**